cpu/stm32/cpu_common.c

```diff
@@ -75,6 +75,27 @@ uint32_t periph_timer_clk(bus_t bus)
     return periph_apb_clk(bus) * apbmul[bus];
 }
 
+void periph_clk_en2(const periph_t *periph)
+{
+    assert(periph);
+
+    const int irq_state = irq_disable();
```

Comment:
the existing `periph_clk_en()` implementation (I believe) is not thread safe. So I have added critical sections to this implementation. Is my thinking correct that this is needed?